### ISSUE TITLE
Randomly interfere with typing text

### ISSUE THREAD

**Issue by fossilet** [tekezo/Karabiner#546]
I enable "use 'hjkl' as arrow keys if you are not editing text", but when I am entering text, e.g. in Spotlight, Help menu search box, and sometimes with the Chinese input method, this feature interferes with text editing such that I cannot enter 'hjkl' as text. However, this does not always happen, it happens randomly with every version up to the latest version 10.15.0.

**Comment by tekezo** [comment]
Is this issue happen only after the first time the Spotlight window is shown?

The Spotlight does not send porperly notification when it is activated.
Therefore, we observe the Spotlight window and detect the activation.

To reduce the observation overhead, there is 10 seconds delay for starting the observation after the first time the Spotlight window is shown.

**Comment by fossilet** [comment]
It is not. I just logged out and activated Spotlight, the inputting worked for hjkl. It is not limited to Spotlight and is really random.

**Comment by tekezo** [comment]
OK. I try to reproduce this issue.

Could you tell me your environment?
https://pqrs.org/osx/karabiner/contact.html.en#required-information

And please show me log messages when this issue happened.
You can confirm the log in Console app.
https://pqrs.org/osx/karabiner/faq.html.en#console

**Comment by fossilet** [comment]
My environment:

OS X version: 10.11.3 Beta (15D9c)
Karabiner version: Karabiner 10.15.0
Your Mac hardware: MacBook Pro (Retina, 15-inch, Mid 2015)
Your keyboard hardware: Builtin keyboard

Pertinent logs:
1/6/16 09:55:34.858 Karabiner[72179]: observer_NSWorkspaceScreensDidWakeNotification
1/6/16 09:55:39.874 Karabiner[72179]: observer_NSWorkspaceDidWakeNotification
1/6/16 10:08:33.839 Karabiner[72179]: observer_NSWorkspaceScreensDidWakeNotification

I can remember that this also happens with previous versions of OS X and Karabiner.

**Comment by tekezo** [comment]
Thank you for information!

I want to see a full log which is not filtered by Karabiner.
Please show me such a full log at the next time you faced this issue.

**Comment by fossilet** [comment]
It also happens sometimes in the search box of Launchpad.

**Comment by tekezo** [comment]
Launchpad is another issue.
We cannot detect the Launchpad window, so Karabiner behavior inherited the previous application.

For example, if Launchpad is opened at using Finder, hjkl will be arrow keys in Launchpad.
If Launchpad is opended at using Safari, hjkl will be normal keys in Launchpad.

**Comment by fossilet** [comment]
@tekezo I have tested it. Thanks for the explanation. Things are complicated.
